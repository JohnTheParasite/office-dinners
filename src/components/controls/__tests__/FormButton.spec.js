import { createLocalVue, shallowMount } from "@vue/test-utils"
import formButton from "@/components/controls/FormButton"
import i18n from "@/i18n"

describe("FormButton", () => {
  const localVue = createLocalVue()
  localVue.use(i18n)

  it("should render button element", () => {
    let buttonComponent = shallowMount(formButton, { localVue, i18n })
    expect(buttonComponent).toMatchSnapshot()
  })

  it("on click method should work", () => {
    let buttonComponent = shallowMount(formButton, { localVue, i18n })
    let buttonClicked = jest.fn()
    buttonComponent.vm.$on("click", buttonClicked)
    buttonComponent.vm.onclick(1)

    expect(buttonClicked).toBeCalledWith(1)
  })

  it("disabled property should work", () => {
    let buttonComponent = shallowMount(formButton, { localVue, i18n, propsData: { disabled: true } })
    expect(buttonComponent).toMatchSnapshot()
  })

  it("loadingInProgress property should work", () => {
    let buttonComponent = shallowMount(formButton, { localVue, i18n, propsData: { loadingInProgress: true } })
    expect(buttonComponent).toMatchSnapshot()
  })

  it("label property should work", () => {
    let buttonComponent = shallowMount(formButton, { localVue, i18n, propsData: { label: "superbutton" } })
    expect(buttonComponent).toMatchSnapshot()
  })

  it("formType property should work", () => {
    let buttonComponent = shallowMount(formButton, { localVue, i18n, propsData: { formType: "button" } })
    expect(buttonComponent).toMatchSnapshot()
  })

  it("type property should work", () => {
    let buttonComponent = shallowMount(formButton, { localVue, i18n, propsData: { type: "primary" } })
    expect(buttonComponent).toMatchSnapshot()
  })

  it("size property should work", () => {
    let buttonComponent = shallowMount(formButton, { localVue, i18n, propsData: { size: "small" } })
    expect(buttonComponent).toMatchSnapshot()
  })
})
