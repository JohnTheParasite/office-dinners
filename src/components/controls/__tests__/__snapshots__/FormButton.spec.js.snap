// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButton disabled property should work 1`] = `
<button disabled="disabled" type="button" class="btn btn-primary">

</button>
`;

exports[`FormButton formType property should work 1`] = `
<button type="button" class="btn btn-primary">

</button>
`;

exports[`FormButton label property should work 1`] = `
<button type="button" class="btn btn-primary">
  superbutton
</button>
`;

exports[`FormButton loadingInProgress property should work 1`] = `
<button type="button" class="btn btn-primary">
  <cssloader-stub></cssloader-stub>
</button>
`;

exports[`FormButton should render button element 1`] = `
<button type="button" class="btn btn-primary">

</button>
`;

exports[`FormButton size property should work 1`] = `
<button type="button" class="btn btn-primary small">

</button>
`;

exports[`FormButton type property should work 1`] = `
<button type="button" class="btn btn-primary">

</button>
`;
